libcontainer/cgroups/fs2/memory.go

```diff
@@ -0,0 +1,103 @@
+// +build linux
+
+package fs2
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"github.com/pkg/errors"
+)
+
+func setMemory(dirPath string, cgroup *configs.Cgroup) error {
+	if cgroup.Resources.MemorySwap != 0 {
+		if err := fscommon.WriteFile(dirPath, "memory.swap.max", strconv.FormatInt(cgroup.Resources.MemorySwap, 10)); err != nil {
+			return err
+		}
+	}
+	if cgroup.Resources.Memory != 0 {
+		if err := fscommon.WriteFile(dirPath, "memory.max", strconv.FormatInt(cgroup.Resources.Memory, 10)); err != nil {
+			return err
+		}
+	}
+
+	// cgroup.Resources.KernelMemory is ignored
+
+	if cgroup.Resources.MemoryReservation != 0 {
+		if err := fscommon.WriteFile(dirPath, "memory.high", strconv.FormatInt(cgroup.Resources.MemoryReservation, 10)); err != nil {
```

Comment:
Hmmmm... I wonder if MemoryReservation should be mapped to `memory.low` instead? @giuseppe what do you think?

The semantics of `memory.high` is similar to `memory.max`, which means start reclaiming more often after going through that threshold (only doesn't trigger OOMs if you're above it.) The semantics of `memory.low` is to *avoid* reclaiming from this cgroup when it's below that threshold (unless the systems is oversubscribed and *needs* to reclaim from this cgroup).

I believe the one we want is the latter one, `memory.min`, right?

Reply:
yes, thanks for noticing it!  I agree `MemoryReservation` should be mapped to `memory.low`

Reply:
updated to use `memory.low`